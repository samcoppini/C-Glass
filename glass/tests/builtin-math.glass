{M
    [m
        (_a)A!
        (_o)(Output)!

        "Testing A.a"(_o)(ol).?
        <513><42>(_a)a.?(_o)(onl).?

        "Testing A.s"(_o)(ol).?
        <513><42>(_a)s.?(_o)(onl).?

        "Testing A.m"(_o)(ol).?
        <513><42>(_a)m.?(_o)(onl).?

        "Testing A.d"(_o)(ol).?
        <267.5><107>(_a)d.?(_o)(onl).?

        "Testing A.f"(_o)(ol).?
        <28.28>(_a)f.?(_o)(onl).?

        "Testing A.lt"(_o)(ol).?
        <100><101>(_a)(lt).?(_o)(onl).?
        <101><101>(_a)(lt).?(_o)(onl).?
        <102><101>(_a)(lt).?(_o)(onl).?

        "Testing A.le"(_o)(ol).?
        <100><101>(_a)(le).?(_o)(onl).?
        <101><101>(_a)(le).?(_o)(onl).?
        <102><101>(_a)(le).?(_o)(onl).?

        "Testing A.e"(_o)(ol).?
        <100><101>(_a)(e).? (_o)(onl).?
        <101><101>(_a)(e).? (_o)(onl).?
        <102><101>(_a)(e).? (_o)(onl).?

        "Testing A.ne"(_o)(ol).?
        <100><101>(_a)(ne).?(_o)(onl).?
        <101><101>(_a)(ne).?(_o)(onl).?
        <102><101>(_a)(ne).?(_o)(onl).?

        "Testing A.ge"(_o)(ol).?
        <100><101>(_a)(ge).?(_o)(onl).?
        <101><101>(_a)(ge).?(_o)(onl).?
        <102><101>(_a)(ge).?(_o)(onl).?

        "Testing A.gt"(_o)(ol).?
        <100><101>(_a)(gt).?(_o)(onl).?
        <101><101>(_a)(gt).?(_o)(onl).?
        <102><101>(_a)(gt).?(_o)(onl).?
    ]
}
