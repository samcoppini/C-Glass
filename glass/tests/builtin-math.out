Testing A.a
555
Testing A.s
471
Testing A.m
21546
Testing A.d
2.5
Testing A.f
28
Testing A.lt
1
0
0
Testing A.le
1
1
0
Testing A.e
0
1
0
Testing A.ne
1
0
1
Testing A.ge
0
1
1
Testing A.gt
0
0
1
